lang/java/avro/src/test/java/org/apache/avro/TestSchemaCompatibility.java

```diff
@@ -260,6 +261,10 @@ public void testUnionReaderWriterSubsetIncompatibility() {
 
       new ReaderWriter(ENUM1_AB_SCHEMA, ENUM1_AB_SCHEMA), new ReaderWriter(ENUM1_ABC_SCHEMA, ENUM1_AB_SCHEMA),
 
+      new ReaderWriter(ENUM1_AB_SCHEMA, ENUM1_AB_SCHEMA),
+
+      new ReaderWriter(ENUM1_AB_SCHEMA, ENUM1_AB_SCHEMA_NAMESPACE),
```

Comment:
This tests the new functionality, but not completely IMHO. Shouldn't we also test these?

* `new ReaderWriter(ENUM1_AB_SCHEMA_NAMESPACE, ENUM1_AB_SCHEMA)`
* `new ReaderWriter(ENUM1_AB_SCHEMA_NAMESPACE2, ENUM1_AB_SCHEMA_NAMESPACE)`

I think we should at least have the first of these. Or does the test in line 266 test the same thing?